### PR TITLE
Update to tree-sitter 0.22.5 and perform recommended upkeep

### PR DESCRIPTION
**Pull request by DavisVaughan**
Closes https://github.com/r-lib/tree-sitter-r/issues/76

Extracted from https://github.com/r-lib/tree-sitter-r/pull/93, I first did a `tree-sitter generate` and committed those new updated files, and then went one by one through the other changes and incorporated most of them.

Stuff to incorporate in follow up PRs, which felt like too much for this upkeep related PR:
- Add in the GitHub Workflows (like publish, fuzz, build) 
- Look into updating `scanner.c` to use `alloc.h` instead of `malloc()` and `array.h` instead of our home grown one

Thanks so much @ObserverOfTime!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,7 +26,7 @@ include = [
 path = "bindings/rust/lib.rs"
 
 [dependencies]
-tree-sitter = "0.21.0"
+tree-sitter = "0.22.5"
```
Comment (DavisVaughan) [inline Cargo.toml:29]: @ObserverOfTime, this update seems to be what fixed the `Cursor` bug https://github.com/tree-sitter/tree-sitter/issues/2767

Do you happen to know which tree-sitter PR it was that fixed it? Just curious!

File: bindings/r/bootstrap.R [hunk 2]
```diff
@@ -1,105 +1,116 @@
...
-  dir_src <- file.path(".", "src")
-  dir_tree_sitter <- file.path(dir_src, "tree_sitter")
+# Update this list if more tree-sitter files are required
+files <- c(
+  file.path("tree_sitter", "alloc.h"),
+  file.path("tree_sitter", "array.h"),
+  file.path("tree_sitter", "parser.h"),
+  "parser.c",
+  "scanner.c"
+)
```
Comment (DavisVaughan) [inline bindings/r/bootstrap.R:8]: Rewrote this to be more maintainable and easier to add to

File: .gitattributes
Comment (DavisVaughan) [inline .gitattributes:1]: Running `tree-sitter test` does show me this. I globally gitignore these but Mac seems to generate them constantly - any way to ignore them upstream?

Screenshot 2024-05-03 at 3 41 15 PM

File: Cargo.toml (hunk 1 above)
Comment (ObserverOfTime) [reply Cargo.toml:29]: No idea.

File: .gitattributes
Comment (ObserverOfTime) [reply .gitattributes:1]: Sure, submit an issue or PR.

File: .gitignore
Comment (ObserverOfTime) [inline .gitignore:1]: You should include the R artifacts here (or the other gitignore file).

```gitignore
# R artifacts
*.Rcheck/
*.tar.gz
```

File: bindings/r/DESCRIPTION
Comment (ObserverOfTime) [inline bindings/r/DESCRIPTION:1]: You should fix the version.

File: bindings/r/DESCRIPTION [hunk 3]
```diff
@@ -18,6 +18,7 @@ Suggests:
 Config/testthat/edition: 3
 Config/build/bootstrap: TRUE
 Encoding: UTF-8
+SystemRequirements: C11
```
Comment (ObserverOfTime) [inline bindings/r/DESCRIPTION:21]: This doesn't do anything. Changing it to `USE_C17` sets `-std=gnu17` which is fine.

File: bindings/r/bootstrap.R
Comment (ObserverOfTime) [inline bindings/r/bootstrap.R:1]: You could add a shebang and make the file executable for ease of use.

**Pull request merged**